php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1402 @@
+          b.firstOf(b.token(STAR), b.token(DIV), b.token(MOD)),
+          CONCATENATION_EXPR()))));
+  }
+
+  public ExpressionTree ADDITIVE_EXPR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.ADDITIVE_EXPR).is(
+      f.additiveExpr(
+        MULTIPLICATIVE_EXPR(),
+        b.zeroOrMore(f.newTuple62(
+          b.firstOf(b.token(PLUS), b.token(MINUS)),
+          MULTIPLICATIVE_EXPR()))));
+  }
+
+  public ExpressionTree SHIFT_EXPR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.SHIFT_EXPR).is(
+      f.shiftExpr(
+        ADDITIVE_EXPR(),
+        b.zeroOrMore(f.newTuple63(
+          b.firstOf(b.token(SL), b.token(SR)),
+          ADDITIVE_EXPR()))));
+  }
+
+  public ExpressionTree RELATIONAL_EXPR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.RELATIONAL_EXPR).is(
+      f.relationalExpr(
+        SHIFT_EXPR(),
+        b.zeroOrMore(f.newTuple64(
+          b.firstOf(b.token(LE), b.token(GE), b.token(LT), b.token(GT)),
+          SHIFT_EXPR()))));
+  }
+
+  public ExpressionTree EQUALITY_EXPR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.EQUALITY_EXPR).is(
+      f.equalityExpr(
+        RELATIONAL_EXPR(),
+        b.zeroOrMore(f.newTuple65(
+          b.firstOf(b.token(NOTEQUAL2), b.token(NOTEQUAL), b.token(EQUAL2), b.token(EQUAL), b.token(NOTEQUALBIS)),
+          RELATIONAL_EXPR()))));
+  }
+
+  public ExpressionTree BITEWISE_AND_EXPR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.BITEWISE_AND_EXPR).is(
+      f.bitwiseAndExpr(
+        EQUALITY_EXPR(),
+        b.zeroOrMore(f.newTuple66(
+          b.token(PHPPunctuator.AMPERSAND),
+          EQUALITY_EXPR()))));
+  }
+
+  public ExpressionTree BITEWISE_XOR_EXPR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.BITEWISE_XOR_EXPR).is(
+      f.bitwiseXorExpr(
+        BITEWISE_AND_EXPR(),
+        b.zeroOrMore(f.newTuple67(
+          b.token(PHPPunctuator.XOR),
+          BITEWISE_AND_EXPR()))));
+  }
+
+  public ExpressionTree BITEWISE_OR_EXPR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.BITEWISE_OR_EXPR).is(
+      f.bitwiseOrExpr(
+        BITEWISE_XOR_EXPR(),
+        b.zeroOrMore(f.newTuple68(
+          b.token(PHPPunctuator.OR),
+          BITEWISE_XOR_EXPR()))));
+  }
+
+  public ExpressionTree LOGICAL_AND_EXPR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.LOGICAL_AND_EXPR).is(
+      f.logicalAndExpr(
+        BITEWISE_OR_EXPR(),
+        b.zeroOrMore(f.newTuple69(
+          b.firstOf(b.token(PHPPunctuator.ANDAND), b.token(PHPKeyword.AND)),
+          BITEWISE_OR_EXPR()))));
+  }
+
+  public ExpressionTree LOGICAL_XOR_EXPR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.LOGICAL_XOR_EXPR).is(
+      f.logicalXorExpr(
+        LOGICAL_AND_EXPR(),
+        b.zeroOrMore(f.newTuple70(
+          b.token(PHPKeyword.XOR),
+          LOGICAL_AND_EXPR()))));
+  }
+
+  public ExpressionTree LOGICAL_OR_EXPR() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)